### PR TITLE
internal IP middleware added

### LINKED ISSUE #218: Securing platform access with IP
In the `settings.py` file, define `INTERNAL_IPS`. Restrict access to the App Panel panel to only those source IPs in `INTERNAL_IPS` when the application is running in a non-development mode.

### PR DESCRIPTION
**Pull request by deepakdinesh1123**
closes #218

### CODE REVIEW

File: backend/src/zcore/apps/shared/platformauth/urls.py [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 from .views import *
 
+from zcore.core.utils import ip_restrict
+
 
 urlpatterns = [
     re_path(
-        r'^login/', 
+        r'^', 
         PlatformUserLoginView.as_view(),
```
Comment (shahharsh176) [inline backend/src/zcore/apps/shared/platformauth/urls.py:14]: for this view we can directly apply the decorator to the view here

File: backend/src/zcore/core/utils.py [hunk 2]
```diff
@@ -98,3 +100,11 @@ def get_search_columns(request):
         if key.startswith("search_"):
             search_columns[key.replace("search_", "")] = value
     return search_columns
+
+def ip_restrict(f):
```
Comment (shahharsh176) [inline backend/src/zcore/core/utils.py:104]: we can rename this to `internal_access_only`

File: backend/src/zcore/core/utils.py [hunk 3]
```diff
@@ -98,3 +100,11 @@ def get_search_columns(request):
...
+    def decorate_view(request):
+        client_ip, is_routable = get_client_ip(request)
+        if not settings.ENV == "dev" and not client_ip in settings.INTERNAL_IPS:
```
Comment (shahharsh176) [inline backend/src/zcore/core/utils.py:107]: we can rewrite condition according to staging and prod env
```python
if settings.ENV in ["staging", "prod"] and client_ip not in settings.INTERNAL_IPS:
    # Deny
# Allow
```

File: backend/src/zcore/middleware/internal_use_only.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from django.urls import reverse, NoReverseMatch
+from django.conf import settings
+
+from django.http import Http404
+from ipware import get_client_ip
+
+
+class InternalUseOnlyMiddleware:
```
Comment (shahharsh176) [inline backend/src/zcore/middleware/internal_use_only.py:8]: to be removed

File: backend/src/zcore/config/urls_public.py [hunk 5]
```diff
@@ -3,22 +3,24 @@
...
 
 urlpatterns = [
     re_path(
         r"^admin/login/$",
-        PlatformUserLoginView.as_view(),
+        decorator_include(ip_restrict, "zcore.apps.shared.platformauth.urls"),
         name="login",
     ),
     re_path(r"^admin/", admin.site.urls),
     re_path(r"^api/", include("zcore.api.platform.urls")),
```
Comment (shahharsh176) [inline backend/src/zcore/config/urls_public.py:20]: we need to add `internal_access_only` decorator here also

File: backend/src/zcore/config/settings/base.py [hunk 6]
```diff
@@ -84,6 +84,7 @@
     "corsheaders.middleware.CorsMiddleware",
     "debug_toolbar.middleware.DebugToolbarMiddleware",
     "zcore.middleware.tenant.TimezoneMiddleware",
+    "zcore.middleware.internal_use_only.InternalUseOnlyMiddleware",
```
Comment (shahharsh176) [inline backend/src/zcore/config/settings/base.py:87]: to be removed now

**Review by shahharsh176 — CHANGES_REQUESTED** [review]

File: backend/src/zcore/config/urls_public.py [hunk 7]
```diff
@@ -3,22 +3,24 @@
...
 from zcore.apps.shared.platformauth.views import (
     PlatformUserLoginView,
 )
+from zcore.core.utils import internal_access_only
 
 urlpatterns = [
     re_path(
         r"^admin/login/$",
-        PlatformUserLoginView.as_view(),
+        decorator_include(internal_access_only, "zcore.apps.shared.platformauth.urls"),
```
Comment (shahharsh176) [inline backend/src/zcore/config/urls_public.py:16]: here we can apply the decorator to direct view, right ?
`internal_access_only(PlatformUserLoginView.as_view())`

**Review by shahharsh176 — CHANGES_REQUESTED** [review]

File: backend/src/zcore/config/urls_public.py (hunk 7 above)
Comment (deepakdinesh1123) [reply backend/src/zcore/config/urls_public.py:16]: yeah, I'll add the change

**Pull request closed without merging**